prov/gni/src/gnix_vc.c

```diff
@@ -892,10 +894,13 @@ static int __gnix_vc_conn_ack_prog_fn(void *data, int *complete_ptr)
 	fastlock_acquire(&ep->vc_ht_lock);
 
 	/*
-	 * we may have already been moved to connecting or
-	 * connected, if so early exit.
+	 * we may have already been moved to connected or
+	 * the datagram for a earlier conn request for this
+	 * vc we are done.  The connection will be completed
```

Comment:
I having trouble parsing this first sentence (especially the "we are done" part).